cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -136,7 +135,7 @@ type Supervisor struct {
 	opampServerPort int
 }
 
-func NewSupervisor(logger *zap.Logger, configFile string) (*Supervisor, error) {
+func NewSupervisor(logger *zap.Logger, cfg config.Supervisor) (*Supervisor, error) {
```

Comment:
is this not tested beyond the e2e test? i'm surprised not to see any tests being updated as part of this PR

Reply:
The Supervisor only recently broke a lot of functionality out of the `NewSupervisor` function that made it challenging to test, but it looks like there's still no unit test covering it. The E2E tests will cover the basics, but a unit test would still be nice for quicker/more granular testing.

Reply:
added 2 small tests